hack/generated/pkg/genruntime/conditions/conditions.go

```diff
@@ -0,0 +1,166 @@
+	Status metav1.ConditionStatus `json:"status"`
+
+	// Severity with which to treat failures of this type of condition.
+	// For conditions which have positive polarity (Status == True is their normal/healthy state), this will be omitted when Status == True
+	// For conditions which have negative polarity (Status == False is their normal/healthy state), this will be omitted when Status == False.
+	// This is omitted in all cases when Status == Unknown
+	// +kubebuilder:validation:Optional
+	Severity ConditionSeverity `json:"severity,omitempty"`
+
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +kubebuilder:validation:Required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// Reason for the condition's last transition.
+	// Reasons are upper CamelCase (PascalCase) with no spaces. A reason is always provided, this field will not be empty.
+	// +kubebuilder:validation:Required
+	Reason string `json:"reason"`
+
+	// Message is a human readable message indicating details about the transition. This field may be empty.
+	// +kubebuilder:validation:Optional
+	Message string `json:"message,omitempty"`
+}
+
+// IsEquivalent returns true if this condition is equivalent to the passed in condition.
+// Two conditions are equivalent if all of their fields EXCEPT LastTransitionTime are the same.
+func (c Condition) IsEquivalent(other Condition) bool {
+	return c.Type == other.Type &&
+		c.Status == other.Status &&
+		c.Severity == other.Severity &&
+		c.Reason == other.Reason &&
+		c.Message == other.Message
+}
+
+// Copy returns an independent copy of the Condition
+func (c Condition) Copy() Condition {
+	return c
+}
+
+// String returns a string representation of this condition
+func (c Condition) String() string {
+	return fmt.Sprintf(
+		"Condition [%s], Status = %q, Severity = %q, Reason = %q, Message = %q, LastTransitionTime = %q",
+		c.Type,
+		c.Status,
+		c.Severity,
+		c.Reason,
+		c.Message,
+		c.LastTransitionTime)
+}
+
+// SetCondition sets the provided Condition on the Conditioner. The condition is only
+// set if the new condition is in a different state than the existing condition of
+// the same type.
+func SetCondition(o Conditioner, new Condition) {
+	if o == nil {
+		return
+	}
+
+	conditions := o.GetConditions()
+	found := false
+	for i := range conditions {
+		existing := conditions[i]
+		if existing.Type == new.Type {
+			found = true
+			if existing.IsEquivalent(new) {
+				// Nothing to do, the conditions are the same
+				return
+			}
+			conditions[i] = new
+			break
+		}
+	}
+
+	if !found {
+		conditions = append(conditions, new)
+	}
+
+	// TODO: do we sort conditions here? CAPI does.
+
+	o.SetConditions(conditions)
+}
+
+// GetCondition gets the Condition with the specified type from the provided Conditioner.
+// If no Condition matches that type, the boolean second parameter is false. If a condition
+// was found, the boolean is true.
```

Comment:
I find these easier to read with the positive case coming first.

```suggestion
// Returns the condition and true if a condition with the specified type is found, or 
// an empty condition and false if not.
```

Reply:
Fixed